s3-get-files.sh

```diff
@@ -8,39 +8,60 @@ log() {
     echo "$(date +"%Y-%m-%d %H:%M:%S") $1" >> "$log_file"
 }
 
-# Get the files in the directory
-log "Listing files in the S3 directory"
-files=$(s3cmd ls s3://infinity-91872 | cut -d/ -f4)
-s3cmd ls s3://infinity-91872 -r | cut -d/ -f4
-log "Getting $files from S3"
+# Create a unique processing directory name with timestamp
+processing_dir="processing_$(date +"%Y-%m-%d_%H-%M-%S")"
+s3cmd mb "s3://infinity-91872/$processing_dir"
+log "Created processing directory: $processing_dir"
 
-s3cmd get "s3://infinity-91872/$files" "/root/SCRIPTS/tmp/"
+# List and filter files, excluding those in processing directories
+files=$(s3cmd ls s3://infinity-91872 | grep -v ' DIR ' | awk '{print $4}' | grep -v '^$')
```

Comment:
Still `print $4` doesn't solve the problem.
For example, there are multiple spaces, then the file name will be the combination of `$4`, `$5`, `$6`, etc.
Consider another approach like `cut -d/ ` to split the whole string into `/`.